lib/recurly/adyen.js

```diff
@@ -0,0 +1,53 @@
+import Emitter from 'component-emitter';
+import errors from '../errors';
+import {Pricing} from './pricing';
+import {normalize} from '../util/normalize';
+
+const debug = require('debug')('recurly:adyen');
+
+/**
+ * Instantiation factory
+ *
+ * @param  {Object} options
+ * @return {Adyen}
+ */
+export function factory (options) {
+  return new Adyen(Object.assign({}, options, { recurly: this }));
+};
+
+
+/**
+ * Initializes an Adyen session.
+ *
+ * @param {Object} options
+ * @constructor
+ * @public
+ */
+
+class Adyen extends Emitter {
+  constructor (options) {
+    super();
+
+    this._ready = false;
+    this.config = {};
+    this.once('ready', () => this._ready = true);
+    this.recurly = options.recurly
+  }
+
+  start (opts) {
+    const payload = {
+      invoiceUuid: opts.invoiceUuid,
+      countryCode: opts.countryCode,
+      shopperLocale: opts.shopperLocale,
+      currencyCode: opts.currencyCode,
+      skinCode: opts.skinCode
+    };
+
+    const frame = this.recurly.Frame({ height: 600, path: '/adyen/start', payload });
+    frame.once('error', cause => this.error('paypal-tokenize-error', { cause }));
```

Comment:
Do these errors cover instances of bad options, like an invalid countryCode, shopperLocal, etc. ? Wondering if we need some kind of validation on those values.

Reply:
Adyen will do its own validations on their end when we try to invoke the modal, although there aren't great errors provided from them around validations. I suppose we could do validations around string length and whatnot here, but not sure how much that will buy us.